src/main/java/com/gregtechceu/gtceu/common/cover/ender/AbstractEnderLinkCover.java

```diff
@@ -0,0 +1,509 @@
+            mainChannelGroup.addWidget(createColorBlockWidget(currentX));
+            currentX += WIDGET_BOARD + 2;
+            mainChannelGroup.addWidget(createConfirmTextInputWidget(currentX));
+
+            mainChannelGroup.addWidget(new ConfirmTextInputWidget(0, WIDGET_BOARD + 2, GROUP_WIDTH - WIDGET_BOARD,
+                    WIDGET_BOARD, cover.getEntry().getDescription(), cover.getEntry()::setDescription,
+                    t -> t == null ? "" : t).setTooltip("cover.ender_fluid_link.tooltip.channel_description"));
+
+            mainGroup.addWidget(mainChannelGroup);
+            mainGroup.addWidget(createWorkingEnabledButton());
+            addEnumSelectorWidgets();
+            mainGroup.addWidget(
+                    cover.addVirtualEntryWidget(cover.getEntry(), 146, WIDGET_BOARD, WIDGET_BOARD, WIDGET_BOARD, true));
+
+            if (cover.getFilterHandler() != null) {
+                mainGroup.addWidget(cover.getFilterHandler().createFilterSlotUI(117, 108));
+                mainGroup.addWidget(cover.getFilterHandler().createFilterConfigUI(10, 72, 156, 60));
+            }
+        }
+
+        @Contract(" -> new")
+        private @NotNull ToggleButtonWidget createToggleButton() {
+            return (ToggleButtonWidget) new ToggleButtonWidget(0, 0, 12, 12, showChannels::getValue, cd -> {
+                showChannels.setValue(!showChannels.getValue());
+                mainGroup.setVisible(showChannels.isFalse());
+                channelsGroup.setVisible(showChannels.isTrue());
+                requestUpdate();
+            }).setTexture(
+                    new GuiTextureGroup(GuiTextures.TOGGLE_BUTTON_BACK.getSubTexture(0, 0, 1, 0.5),
+                            GuiTextures.BUTTON_LIST),
+                    new GuiTextureGroup(GuiTextures.TOGGLE_BUTTON_BACK.getSubTexture(0, 0.5, 1, 0.5),
+                            GuiTextures.BUTTON_LIST))
+                    .setHoverTooltips("cover.ender_fluid_link.tooltip.list_button");
+        }
+
+        @Contract("_ -> new")
+        private @NotNull Widget createToggleButtonForPrivacy(int currentX) {
+            return new EnumSelectorWidget<>(currentX, 0,
+                    WIDGET_BOARD, WIDGET_BOARD, Permissions.values(), cover.permission, cover::setPermission);
+        }
+
+        private ColorBlockWidget createColorBlockWidget(int currentX) {
+            return new ColorBlockWidget(currentX, 0, WIDGET_BOARD, WIDGET_BOARD).setColorSupplier(cover::getColor);
+        }
+
+        private ConfirmTextInputWidget createConfirmTextInputWidget(int currentX) {
+            int GROUP_X = 10;
+            int textInputWidth = (GROUP_WIDTH - GROUP_X * 2) - currentX - WIDGET_BOARD - 2;
+            return new ConfirmTextInputWidget(currentX, 0, textInputWidth, WIDGET_BOARD, cover.colorStr,
+                    cover::setChannelName, text -> {
+                        if (text == null || !COLOR_INPUT_PATTERN.matcher(text).matches() || text.length() > 8) {
```

Comment:
...and then you can also remove the `len > 8` check here.
```suggestion
                        if (text == null || !COLOR_INPUT_PATTERN.matcher(text).matches()) {
```